featureDevelopmentGotchas.md

```diff
@@ -0,0 +1,25 @@
+# Edge Case Spidey Sense Tips and Tricks - Things to keep in mind for code development/reviews:
+(aka this is the kind of shit that has bitten us in the past)
+
+## Future proof?
+- Are you building in convention instead of design?
+  - How easy would it be for an unfamiliar dev to use your code improperly?
+
+## Complete logic
+- Are there unaccounted for logical branches?
+  - Wherever there is an if, consider the else
```

Comment:
I might make a PR into this after this is merged for the Swift specific `guard` syntax, mostly stating the same thing.